design/design-doc-observe-only-resources.md

```diff
@@ -0,0 +1,293 @@
+ # Observe Only Resources
+
+* Owners: Hasan Turken (@turkenh)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+When using Crossplane to manage resources, we typically create a Managed resource that represents the desired state of
+the resource at a provider. Crossplane takes ownership of the resource, starts acting as the source of truth, and
+ensures configuration matches the desired state.
+
+Sometimes, we want to "observe" an existing resource without taking ownership of it. This could be useful in several
+scenarios, which could be grouped as follows:
+
+- Referencing existing resources without managing them
+  - In your composition, you want to reference network resources like VPC and subnets that are managed by another tool
+  or team.
+- Fetching data from existing resources
+  - You need information about an existing VPC, such as its CIDR range and the subnets it contains.
+- Gradual migration of existing/legacy infrastructure to Crossplane
+  - You have existing infrastructures managed by Terraform, and you want to migrate them gradually to Crossplane.
+  - You have a legacy infrastructure that you want to migrate to Crossplane, but you want to test the managed resources
+  before taking ownership of the underlying resources.
+  - For an existing resource, you don’t want to give some initial configuration that might override the actual
+  configuration. You want to late-initialize all fields, including the ones that would be required otherwise.
```

Comment:
Sure, I have a VPC created by click ops. I want to migrate it to Crossplane. 
Currently, I need to create a VPC with same configuration of your existing resource so that Crossplane does not change its configuration. This is trying to say, you create with observe only without giving any parameter, e.g. [cidrBlock](https://marketplace.upbound.io/providers/crossplane-contrib/provider-aws/v0.33.0/resources/ec2.aws.crossplane.io/VPC/v1beta1#doc:spec-forProvider-cidrBlock), verify the configuration filled in including cidrBlock, change policy to fullControl.

Realizing this is more of an issue with how we import today, where you need to provide required fields but assuming we solved that with CEL, starting with ObserveOnly gives you a chance to verify all look good before giving full control to XP.

Reply:
Rephrased the statement in the doc.